### ISSUE TITLE
Fails to parse 2.x print statements

### ISSUE THREAD

**Issue by onelson** [ambv/black#108]
Mentioned briefly on [twitter](https://twitter.com/theomn/status/981596730696245248)

Given a 2.7 codebase with print statements in it, such as this `bad.py`:

```python

def plain_print(arg):
    """simple function to print things plainly"""


    print arg
```
`black` reports the following:
```
black bad.py 
error: cannot format bad.py: cannot use --safe with this file; failed to parse source file with Python 3.6's builtin AST. Re-run with --fast or stop using deprecated Python 2 syntax. AST error message: Missing parentheses in call to 'print'. Did you mean print(arg)? (<unknown>, line 7)
```
Everything it says is correct. I could stop using deprecated Python 2 syntax (by importing `print_function` from `__future__`, or worse by pretending the statement is a function by wrapping the arg in parens). I could also elect to use `--fast`, which also seems to work.

Still, it would be nice if this _just worked_ for 2.7 sources.

----
Python version: 3.6
Black version: 18.4a0

**Comment by ambv** [comment]
Let me know how I can improve the error message here since I think it says exactly what went wrong here.

You're using a piece of syntax that doesn't parse with Python 3's AST. I'm not using this AST for formatting but I *am* using it later to confirm that code "before" and "after" still looks the same to Python. I can't use lib2to3's CST for this because "before" and "after" in this case is obviously different.

In other words, Black **is** able to reformat your file just fine, it just cannot confirm for you that it didn't make any errors while doing it. Yes, it would be nice if that worked to but it would require a lot of work for something that can be trivially rectified on the user's end by using print functions.

I hoped the error message explained this in a bit more succinct language.

**Comment by ambv** [comment]
Hm. How would you feel if Black reformatted print statements to be print functions instead?

**Comment by onelson** [comment]
I actually think that'd be grand (if possible). I can see that getting tricky for cases where folks are using `print` with non-standard file object via `>>`, but if it's possible then I'd say go for it.

Is this the only breaking piece of syntax between 2 and 3?

**Comment by ambv** [comment]
Alright, so we'll do that. There is also `exec` which is unfortunately harder to fix (requires `six` or `future`) so I'm not going to touch that.

**Comment by buddhhu** [comment]
`failed to parse source file.  AST error message: expression cannot contain assignment, perhaps you meant "=="? (<unknown>, line 73)`

I'm getting this even if there is no error.

`async def download_episode(download_url):
    anime_name = name_parser(download_url)
    filename = anime_name + download_url[-4:]
    make_directory(anime_name)
    download_path = os.path.join(anime_name, filename)
    if not os.path.exists(download_path):
        await (
            await asyncio.create_subprocess_shell(
                f'youtube-dl "{download_url}" -o "{download_path}"',
                stdout=asyncio.subprocess.PIPE,
                stderr=asyncio.subprocess.PIPE,
            )
        ).communicate()
        clear_tmp(anime_name)
    return download_path`

Last line of code in line 74
Python version: 3.9.6

**Comment by JelleZijlstra** [comment]
@buddhhu could you open a new issue for that with a complete code sample? I'm unable to reproduce a crash on the code you posted.

**Comment by buddhhu** [comment]
> @buddhhu could you open a new issue for that with a complete code sample? I'm unable to reproduce a crash on the code you posted.

Done #2403
